framework/src/utils/SlepcSupport.C

```diff
@@ -115,13 +207,7 @@ storeSlepcOptions(FEProblemBase & fe_problem, const InputParameters & params)
 
   if (params.isParamValid("solve_type"))
   {
-    mooseError("Can not use \"solve_type\" for eigenvalue problems, please use "
-               "\"eigen_solve_type\" instead \n");
-  }
-
-  if (params.isParamValid("eigen_solve_type"))
-  {
-    const std::string & eigen_solve_type = params.get<MooseEnum>("eigen_solve_type");
+    const std::string & eigen_solve_type = params.get<MooseEnum>("solve_type");
```

Comment:
Are you sure you want to retrieve this as a string? It's stored as a MooseEnum type which is better.